File: Python/list_node/list_node.py
class ListNode:
    def __init__(self, value = 0, next = None):
        self.value = value
        self.next = next

    def __eq__(self, o: object) -> bool:
        this = self
        if o.__class__ != ListNode:
            return False
        while this is not None:
            if o is None:
                return False
            if o.value != this.value:
                return False
            this = this.next
            o = o.next
        return True

    def __str__(self):
        return f'(value={self.value}, next={self.next})'

File: Java/src/main/java/com/home/algorithms/leetcode/listnode/ListNode.java
package com.home.algorithms.leetcode.listnode;

import java.util.Objects;

public class ListNode {
    int val;
    ListNode next;

    public ListNode(int x) {
        val = x;
    }

    ListNode(int val, ListNode next) { this.val = val; this.next = next; }

    public String toString() {
        ListNode ln = this;
        StringBuilder sb = new StringBuilder("[");
        while (ln != null) {
            sb.append(ln.val).append(",");
            ln = ln.next;
        }
        return sb.replace(sb.length() - 1, sb.length(), "").append("]").toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ListNode listNode = (ListNode) o;
        return val == listNode.val && Objects.equals(next, listNode.next);
    }

    @Override
    public int hashCode() {
        return Objects.hash(val, next);
    }
}
